blockchain/types/block_test.go

```diff
@@ -194,6 +193,75 @@ func TestBlockEncoding(t *testing.T) {
+	var block Block
+	if err := rlp.DecodeBytes(blockEnc, &block); err != nil {
+		t.Fatal("decode error: ", err)
+	}
+	//
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, want %v", f, got, want)
+		}
+	}
+
+	check("ParentHash", block.ParentHash(), b.ParentHash())
+	check("Rewardbase", block.Rewardbase(), b.Rewardbase())
+	check("Root", block.Root(), b.Root())
+	check("TxHash", block.TxHash(), b.TxHash())
+	check("ReceiptHash", block.ReceiptHash(), b.ReceiptHash())
+	check("Bloom", block.Bloom(), b.Bloom())
+	check("BlockScore", block.BlockScore(), b.BlockScore())
+	check("NUmber", block.Number(), b.Number())
+	check("GasUsed", block.GasUsed(), b.GasUsed())
+	check("Time", block.Time(), b.Time())
+	check("TimeFoS", block.TimeFoS(), b.TimeFoS())
+	check("Extra", block.Extra(), b.Extra())
+	check("Hash", block.Hash(), b.Hash())
+	check("Size", block.Size(), common.StorageSize(len(blockEnc)))
+
+	// Create legacy tx.
+	to := common.HexToAddress("095e7baea6a6c7c4c2dfeb977efac326af552d87")
+	tx1 := NewTx(&TxInternalDataLegacy{
+		AccountNonce: 0,
+		Recipient:    &to,
+		Amount:       big.NewInt(10),
+		GasLimit:     50000,
+		Price:        big.NewInt(10),
+	})
+	sig := common.Hex2Bytes("9bea4c4daac7c7c52e093e6a4c35dbbcf8856f1af7b059ba20253e70848d094f8a8fae537ce25ed8cb5af9adac3f141af69bd515bd2ba031522df09b97dd72b100")
+	tx1, _ = tx1.WithSignature(LatestSignerForChainID(big.NewInt(1)), sig)
+
+	// Create ACL tx.
+	addr := common.HexToAddress("0x0000000000000000000000000000000000000001")
+	tx2 := NewTx(&TxInternalDataAccessList{
+		ChainID:      big.NewInt(1),
+		AccountNonce: 0,
+		Recipient:    &to,
+		GasLimit:     123457,
+		Price:        big.NewInt(10),
+		AccessList:   AccessList{{Address: addr, StorageKeys: []common.Hash{{0}}}},
+	})
+	sig2 := common.Hex2Bytes("3dbacc8d0259f2508625e97fdfc57cd85fdd16e5821bc2c10bdd1a52649e8335476e10695b183a87b0aa292a7f4b78ef0c3fbe62aa2c42c84e1d9c3da159ef1401")
+	tx2, _ = tx2.WithSignature(LatestSignerForChainID(big.NewInt(1)), sig2)
+	//
+	check("len(Transactions)", len(block.Transactions()), 2)
+	check("Transactions[0].Hash", block.Transactions()[0].Hash(), tx1.Hash())
+	check("Transactions[1].Hash", block.Transactions()[1].Hash(), tx2.Hash())
+	check("Transactions[1].Type()", block.Transactions()[1].Type(), TxTypeAccessList)
+
+	ourBlockEnc, err := rlp.EncodeToBytes(&block)
+	if err != nil {
+		t.Fatal("encode error: ", err)
+	}
+	if !bytes.Equal(ourBlockEnc, blockEnc) {
+		t.Errorf("encoded block mismatch:\ngot:  %x\nwant: %x", ourBlockEnc, blockEnc)
+	}
```

Comment:
I cannot understand the purpose of this test. `block` has decoded values of `blockEnc`. Therefore, `rlp.EncodeToBytes(&block)` should be the same with `blockEnc`.

Reply:
It was tested to see if the block containing AccessList tx was binary encoded. Do you think it's not necessary? Then I will delete it.

Reply:
Then maybe you can create another block which includes above txs to test encoding. 
`block` has decoded values from `blockEnc`, so encode `block` again to test equal with `blockEnc` feels for sure thing.